andela/ah-infinity-stones
# #161966934 Implement reporting of an article

#### What does this PR do?
- Implement reporting of an article functionality.
#### Description of Task to be completed?
- Write article reporting tests. They should be failing.
- Create an endpoint for reporting an individual article
- Implement article reporting functionality. A mail is sent to admin
- Make the tests pass.
#### How should this be manually tested?
1. `git clone -b ft-users-able-report-article-161966934 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='authors_haven'
export DATABASE_USER='postgres'
export DATABASE_PASSWORD='postgres'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='test_authors_haven'
export EMAIL_HOST_USER='Infinites'
export EMAIL_HOST_PASSWORD='st1fflers'
export SECRET_KEY='my secret my choice'
export DATABASE_URL="postgresql://localhost/authors_haven"
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
{ "user": {
"username": "your name",
"email": "your@email.com",
"password": "@Password321"	
}
}
```
Check your email and activate the account. Use the registration token to access the following endpoints:
- `http://127.0.0.1:8000/api/articles` - Post an article. 
```
{
    "title": "How to make it rain",
    "description": "Ever wonder how?",
    "body": "You have to pray",
    "tag": ["reactjs", "angularjs", "rain"]
  
}
```
-  To report an article, copy and paste the slug ( in this case `how-to-make-it-rain`) into the url and make a POST request to `http://127.0.0.1:8000/api/articles/<slug>/report` (eg. ` http://127.0.0.1:8000/api/articles/how-to-make-it-rain/rate` ) with the following data:
```
{
    "report_msg": "This has been plagiarised from my site."
}
```


8. Run tests`python3 manage.py test authors.apps.articles.tests.test_views.ArticleReportingTestCase`
#### What are the relevant pivotal tracker stories?
[#161966934](https://www.pivotaltracker.com/story/show/161966934)
#### Screenshots
<img width="1123" alt="screenshot 2018-12-21 at 11 19 46" src="https://user-images.githubusercontent.com/16904946/50332505-5e880200-0513-11e9-848a-8d98889f9e4b.png">
<img width="1121" alt="screenshot 2018-12-21 at 11 28 07" src="https://user-images.githubusercontent.com/16904946/50332564-8ecfa080-0513-11e9-9089-5d125636f41b.png">
<img width="1118" alt="screenshot 2018-12-21 at 11 23 15" src="https://user-images.githubusercontent.com/16904946/50332575-9c852600-0513-11e9-8fb7-8e2938bf6cc9.png">
<img width="532" alt="screenshot 2018-12-21 at 11 24 19" src="https://user-images.githubusercontent.com/16904946/50332585-a3ac3400-0513-11e9-806e-944812c75824.png">

<img width="997" alt="screenshot 2018-12-19 at 20 33 54" src="https://user-images.githubusercontent.com/16904946/50332594-ad359c00-0513-11e9-94d6-930687fe1ed6.png">
